ci-automation/vendor-testing/aws.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# Copyright (c) 2022 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+# Test execution script for the AWS vendor image.
+# This script is supposed to run in the mantle container.
+
+source ci-automation/vendor_test.sh
+
+board="${CIA_ARCH}-usr"
+image_name="ci-${CIA_VERNUM}"
+aws_instance_type_var="AWS_${CIA_ARCH}_INSTANCE_TYPE"
+aws_instance_type="${!instance_type_var}"
+more_aws_instance_types_var="AWS_${CIA_ARCH}_MORE_INSTANCE_TYPES"
+more_aws_instance_types=( ${!more_instance_types_var} )
+
+if [[ "${AWS_AMI_ID}" == "" ]]; then
+    echo "++++ ${CIA_TESTSCRIPT}: downloading flatcar_production_ami_vmdk_image.vmdk.bz2 for ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+    copy_from_buildcache "images/${CIA_ARCH}/${CIA_VERNUM}/flatcar_production_ami_vmdk_image.vmdk.bz2" .
+
+    lbunzip2 'flatcar_production_ami_vmdk_image.vmdk.bz2'
+
+    AWS_BUCKET="flatcar-kola-ami-import-${AWS_REGION}"
+    trap 'ore -d aws delete --region="${AWS_REGION}" --name="${image_name}" --ami-name="${image_name}" --file="flatcar_production_ami_vmdk_image.vmdk" --bucket "s3://${AWS_BUCKET}/${board}/"' EXIT
+    ore aws initialize --region="${AWS_REGION}" --bucket "${AWS_BUCKET}"
+    AWS_AMI_ID=$(ore aws upload --force --region="${AWS_REGION}" --name="${image_name}" --ami-name="${image_name}" --ami-description="Flatcar Test ${image_name}" --file='flatcar_production_ami_vmdk_image.vmdk' --bucket  "s3://${AWS_BUCKET}/${board}/" | jq -r .HVM)
+    echo "++++ ${CIA_TESTSCRIPT}: Created new AMI ${AWS_AMI_ID} (will be removed after testing) ++++"
+fi
+
+run_aws_kola_test() {
+    local instance_type="${1}"; shift
+    local instance_tapfile="${1}"; shift
+
+    timeout --signal=SIGQUIT 6h \
+        kola run \
+         --board="${board}" \
+         --basename="${image_name}" \
+         --channel="${CIA_CHANNEL}" \
+         --offering='basic' \
+         --parallel="${AWS_PARALLEL}" \
+         --platform=aws \
+         --aws-ami="${AWS_AMI_ID}" \
+         --aws-region="${AWS_REGION}" \
+         --aws-type="${instance_type}" \
+         --aws-iam-profile="${AWS_IAM_PROFILE}" \
+         --tapfile="${instance_tapfile}" \
+         --torcx-manifest=../torcx_manifest.json \
+         "${@}"
+}
+
+# these are set in ci-config.env
+export AWS_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY
+
+extra_tests_run=0
+for instance_type in "${more_aws_instance_types[@]}"; do
+    instance_tests=( $(filter_prefixed_tests "${instance_type}" "${@}") )
```

Comment:
I don't really follow, either this is generic for any test case or it's only for the `cl.internet` test which is why it has a special treatment below. Right now this seems like a mix of both but not really generic, so I don't think we should pretend it is?

Reply:
Yeah, I'm not yet happy with this part. Wanted to make it generic in case we wanted to run more tests, or a different test on the additional instances and it should be an easy change to make.